week-2-michael-rogers/code-challenges.test.js

```diff
@@ -0,0 +1,170 @@
+// ASSESSMENT 2: Coding Practical Questions with Jest
+
+// Please read all questions thoroughly
+// Pseudo coding is REQUIRED
+// If you get stuck, please leave comments to help us understand your thought process
+
+// Use test driven development to complete the following questions
+// Add appropriate dependencies: $ yarn add jest
+
+// Reminder: The test will call your function
+// Run the file with the following command: $ yarn jest
+
+// --------------------INSTRUCTOR EXAMPLE: Create a function that takes in an array of numbers and returns an array with all the numbers multiplied by 3.
+
+// a) Create a test with expect statements for each of the variables provided.
+
+const numbersArray1 = [6, 7, 8, 9, 10]
+// Expected output: [18, 21, 24, 27, 30]
+const numbersArray2 = [24, 27, 30, 33, 36]
+// Expected output: [72, 81, 90, 99, 108]
+
+// b) Create the function that makes the test pass.
+
+//Reusable test template
+
+// describe("greeter", () => {
+//     it("returns a generic greeting", () => {
+//       expect(greeter()).toEqual("Hello, LEARN student!")
+//     })
+//   })
+
+// --------------------1) Create a function that takes a object as an argument and decides if the number inside it is evenly divisible by three or not.
+
+// a) Create a test with expect statements for each of the variables provided.
+
+const object1 = { number: 15 }
+// Expected output: "15 is divisible by three"
+const object2 = { number: 0 }
+// Expected output: "0 is divisible by three"
+const object3 = { number: -7 }
+// Expected output: "-7 is not divisible by three"
+//const test = true //Test variable to try and throw an error
+//Expected output: "Error: Function input is not an object or key-pair value is not a number"
+
+describe("divisbleByThree", () => {
+    //console.log(test)
+    it("returns an appropriate message if the number inside the object is divisible by 3 or not", () => {
+      expect(divisibleByThree(object1)).toEqual(`${15} is divisible by three`)
+      expect(divisibleByThree(object2)).toEqual(`${0} is divisible by three`)
+      expect(divisibleByThree(object3)).toEqual(`${-7} is not divisible by three`)
+      //expect(divisibleByThree(test)).toEqual("Error: Function input is not an object or key-pair value is not a number")
+        //Had difficulty trying to replicate the error message in my test. Function expects an object, so I tested by passing in a regular variable with a boolean
```

Comment:
I wish I had a screenshot of that being passed and the error that came back with it, I would have loved to see it. If it passed from just running your test again, it could have been a false true, or if it was the console.log that passed it I would have to have seen it.